jwala-services/src/main/java/com/cerner/jwala/service/media/impl/MediaServiceImpl.java

```diff
@@ -113,16 +120,28 @@ public void remove(final String name) {
      *
      * @param name media name
      */
-    private void checkForJvmAssociation(String name) {
+    private void checkForAssociation(String name, MediaType type) {
         List<Jvm> jvmList = jvmPersistenceService.getJvms();
+        List<WebServer> webServerList = webServerPersistenceService.getWebServers();
         List<String> existingAssosiations = new ArrayList<>();
         for (Jvm jvm : jvmList) {
-            if (jvm.getJdkMedia() != null && name.equalsIgnoreCase(jvm.getJdkMedia().getName())) {
-                existingAssosiations.add(jvm.getJvmName());
+            if (jvm.getJdkMedia().getType() == type || jvm.getTomcatMedia().getType() == type) {
```

Comment:
Personally, I would group all of the jdkMedia and jdkTomcat conditional tests together. For instance, if the type is Tomcat, the code will still check for jdkMedia and jdkTomcat - and now that I think about it, won't this add existing associations more than once? I would change existingAssociations to a Set as well to avoid duplicates.